File: Blog.ReactUI/Src/Types/RecordDTO.ts
import RecordStateDTO from "Types/RecordStateDTO";
import UserDTO from "Types/UserDTO";
import CommentDTO from "Types/CommentDTO";

export default class RecordDTO {
  public RecordId: number;

  public Name: string;

  public Content: string;

  public CreateDate: Date;

  public State: RecordStateDTO;

  public Comments: CommentDTO[];

  public UserId: number;

  public User: UserDTO;
}

File: Blog.Services/Models/RecordDTO.cs
using Blog.Services.Enums;
using Blog.Services.Identity.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Blog.Services.Models
{
    public class RecordDTO
    {
        public int RecordId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Content { get; set; }

        public DateTime CreateDate { get; set; }

        public RecordStateDTO State { get; set; } = RecordStateDTO.Private;

        public ICollection<CommentDTO> Comments { get; set; } = new List<CommentDTO>();

        public int UserId { get; set; }

        public UserDTO User { get; set; }
    }
}
